external_parser/joiners/example_joiner.cc

```diff
@@ -482,6 +484,10 @@ bool example_joiner::process_joined(v_array<example *> &examples) {
     return false;
   }
 
+  if(_binary_to_json) {
```

Comment:
do we now need to remove lines 502 to 505 which also check `_binary_to_json`?

Reply:
actually I am confused, without the calc_and_set_reward, how are we calculating the reward?

Reply:
I thought the calculation was done on log_converted as it works for original_reward.
The issue is that `calc_and_set_reward` needs to be split in two, one o calc and one to set. The calc part is called always and set only when !binary_to_json.

Working on it

Reply:
split done. it now works as suggested.